source/shared/cpp/ObjectModel/UnknownElement.cpp

```diff
@@ -13,9 +8,19 @@ UnknownElement::UnknownElement() : BaseCardElement(CardElementType::Unknown)
 {
 }
 
+Json::Value UnknownElement::SerializeToJsonValue() const
+{
+    // Important -- we need override the type value that the base element uses for us
+    Json::Value root = BaseCardElement::SerializeToJsonValue();
+    root[AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::Type)] = GetElementTypeString();
+    return root;
+}
+
 std::shared_ptr<BaseCardElement> UnknownElementParser::Deserialize(ParseContext&, const Json::Value& json)
 {
+    std::string actualType = ParseUtil::GetTypeAsString(json);
```

Comment:
Maybe add a comment here explaining why we need this.

Reply:
Good idea :)

---
In reply to: [251154183](https://github.com/Microsoft/AdaptiveCards/pull/2332#discussion_r251154183) [](ancestors = 251154183)